### ISSUE TITLE
[SF][A11y][List] Unique accessible names are not provided for checkboxes

### ISSUE THREAD

**Issue by saorabhkr** [SAP/ui5-webcomponents#6861]
### Describe the bug

For a List/Tree where we have `MultiSelect` enable we observed unique accessible names are not there for the checkboxes

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/iframe.html?args=mode:MultiSelect&id=data-display-list--default&viewMode=story

### Reproduction steps

Go to [List Page](https://sap.github.io/ui5-webcomponents-react/iframe.html?args=mode:MultiSelect&id=data-display-list--default&viewMode=story)

Enable JAWS shortcuts keys
Try accessing the checkboxes, and open JAWS shortcuts keys dialog will not have a unique name for checkboxes. 

### Expected Behaviour

For a List/Tree where we have `MultiSelect` enable all the checkboxes should have a unique names. a unique name

### Screenshots or Videos

![MicrosoftTeams-image (8)](https://user-images.githubusercontent.com/5204425/229699306-18415f4a-0dda-4cc8-a506-3442072d5434.png)

### UI5 Web Components for React Version

v1.12.0

### UI5 Web Components Version

v1.11.0

### Browser

Chrome

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by yanaminkova** [comment]
Hello @saorabhkr,

Thanks for reporting.

I am not sure that I fully understand the issue.
In the example you have provided, the `Checkboxes` do have `accessible-names`, which are being announced by JAWS and also visible on the Virtual HTML Features Dialog, which screenshot you have also added in your message.

Could you please give a bit more information and steps for reproducing the reported problem so we could help you.

Thanks and Kind Regards,
Yana

**Comment by saorabhkr** [comment]
> Hello @saorabhkr,
> 
> Thanks for reporting.
> 
> I am not sure that I fully understand the issue. In the example you have provided, the `Checkboxes` do have `accessible-names`, which are being announced by JAWS and also visible on the Virtual HTML Features Dialog, which screenshot you have also added in your message.
> 
> Could you please give a bit more information and steps for reproducing the reported problem so we could help you.
> 
> Thanks and Kind Regards, Yana

Hello @ yanaminkova,

Yes you are absolutly right, `checkboxes` have `accessible-names` but if try navigation with JAWS keybord help it is printing common name, it should be unique. 

To open JAWS keybord help,  press and hold INSERT and press F1 twice quickly. Once JAWS help is open, press F6 to move to the help topic window. Form that window select `checkbox` and observe the output. Please do let me knwo If you need any further help. Hope this helps. 

Best Regards, 
Saorabh

**Comment by dobrinyonkov** [comment]
We've discussed this question with the central accessibility team and we await for their feedback.
I am setting this one to blocked until we receive a response.

**Comment by dobrinyonkov** [comment]
Hi @GongRichard,

we've explored the possible solutions here: 

1) fordward the `accessible-name` of the list item to the checkbox - as long as apps provide uniqe accessible names to the items, the checkboxes will also have such.

2) internally set the text content of the list items as label of the checkbox.

3) hide the checkboxes.

There isn't really a general recommendation from the ARIA specification about such pattern. 
We've submitted a question to the aria practices and patterns repository asking to comment on this.

Let me know if you have any thoughts on the above.

**Comment by dobrinyonkov** [comment]
Hi @GongRichard, 

After thorough discussions with accessibility experts, we have reached a point that assistive technologies (AT) may autonomously handle the labeling of such cases, as they possess the required contextual understanding.  

It needs to be clarified why the labelling in closed context should be handled by web authors or web control developers and not by the user agent or assistive technology (auto-labelling). Assistive technologies are very aware of the context they consume from platform accessibility trees and could provide in their speech output by extended element info triggered by a special key or in their virtual overview dialogs means to identify the very position of e.g. a checkbox in a list/table cell with row/col position and list/table header information.

The opposite would mean, to make the labels really globally unique each and every such control (graphics, inputs, links, buttons, etc.) must be additionally labelled with the list header (there could be multiple list on the screen), the list item info (index / row header) and the column info if we’re inside a table (index / column header). The screenreader output would be horror. 

More information and further discussions can be tracked with this GitHub issue: https://github.com/w3c/aria-practices/issues/2810

Kind Regards,
Dobrin

**Comment by dobrinyonkov** [comment]
After consulting with accessibility experts, we believe that assistive technologies (AT) are best suited to handle labeling in such cases, as they can provide contextual understanding autonomously. We feel this should be managed by the AT, not web authors or developers, to avoid overwhelming users with redundant information.

Best regards,
Dobrin

**Comment by dobrinyonkov** [comment]
More discussion about this will take place next week, I will post updates after.

**Comment by GongRichard** [comment]
Hi @dobrinyonkov，
Any progress about this issue? We have customer to complain this issue.
Thanks,
Richard

**Comment by yanaminkova** [comment]
Hello @GongRichard,

The outcome of the discussions was that we cannot offer a solution at the moment.

Referring to our central note: https://me.sap.com/notes/3260235, the component is to be considered conformant as per the definition of the W3C.

More specifically, the following sections:
 
3) Primary and Secondary Functions of ATs
As per W3C definition, there needs to be one conforming version of a web page. SAP follows this recommendation by providing a code base that implements up-to-date accessibility standards (see point 1 of this section). Should ATs prove SAP’s code base viable through their Main Functionality, such as speech output, but render it invalid through a Secondary Means, such as go-to-functions, then the code base is to be considered conformant as per the definition of the W3C.
 
4) Fixing Issues in Secondary Functionalities of ATs
Issues in Secondary Functionalities of ATs will be fixed only if they are reflected in both, the SAP codebase and the ATs' Primary Funtionality.
 
as well as the following:
 
2.5.3 - Label in Name
This Success Criterion requires that the text of the visible label is part of the (invisible) accessible name. It does not require that the accessible name is unique. Instead, it shows that the accessible name can be equal to the label, which in return does not need to be unique.

I will therefore close this case.
Thank you for understanding!

Kind regards,
Yana

**Comment by judith9209** [comment]
disagree in above explanation. The WCAG issue 2810 should not have been taken for this issue as an answer. It describes different topics. For this issue, the accessible name of a checkbox is "Multiple Selection mode" that does not match the visible name of each checkbox "List Item 1".  
You are referring to 2.5.3 - Label in Name that is full of quotes why this issue can be perceived as valid because visible text does not match accessible name:

https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html
"Label in Name (Level A)"
"The intent of this Success Criterion is to ensure that the words which visually label a component are also the words associated with the component programmatically. This helps ensure that people with disabilities can rely on visible labels as a means to interact with the components."
"Users typically have a much better experience if the words and characters in the visible label of a control match or are contained within the accessible name. "
"Where text labels exist and are properly linked to the user interface components through established authoring practices, the label and name will normally match. When they don't match, speech-input users who attempt to use the visible text label as a means of navigation or selection (e.g., "move to Password") will be unsuccessful. The speech-based navigation fails because the visible label spoken by the users does not match (or is not part of) the accessible name that is enabled as a speech-input command. In addition, when the accessible name is different from the visible label, it may function as a hidden command that can be accidentally activated by speech-input users."
"Mismatches between visible labels and programmatic names for controls are even more of an issue for speech-input and text-to-speech users who also have cognitive challenges."
https://www.w3.org/WAI/WCAG21/Techniques/general/G211
"Matching the accessible name to the visible label"
https://www.w3.org/WAI/WCAG21/Techniques/general/G208
"Including the text of the visible label as part of the accessible name"
"the text string that makes up the visible label must occur in its entirety in the accessible name. "

**Comment by judith9209** [comment]
Feel free to have a look at the wcag discussion about this issue: https://github.com/w3c/wcag/discussions/4190
